--- pvemanagerlib.js.orig	2019-03-28 13:50:36.755001263 +0300
+++ pvemanagerlib.js	2019-04-08 11:52:49.344119808 +0300
@@ -6002,7 +6002,8 @@
 	['comstar', 'Comstar'],
 	[ 'istgt', 'istgt'],
 	[ 'iet', 'IET'],
-	[ 'LIO', 'LIO']
+	[ 'LIO', 'LIO'],
+	[ 'ctld', 'CTLD']
     ]
 });
 Ext.define('PVE.form.DayOfWeekSelector', {
@@ -30735,6 +30736,7 @@
 	data: {
 	    isLIO: false,
 	    isComstar: true,
+        isCTLD: false,
 	    hasWriteCacheOption: true
 	}
     },
@@ -30750,6 +30752,7 @@
 	    var vm = this.getViewModel();
 	    vm.set('isLIO', newVal === 'LIO');
 	    vm.set('isComstar', newVal === 'comstar');
+	    vm.set('isCTLD', newVal === 'ctld');
 	    vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'istgt');
 	}
     },
@@ -30849,7 +30852,7 @@
 		xtype: me.isCreate ? 'textfield' : 'displayfield',
 		name: 'lio_tpg',
 		value: '',
-		bind: me.isCreate ? { disabled: '{!isLIO}' } : { hidden: '{!isLIO}' },
+		bind: me.isCreate ? { disabled: '{!isCTLD}' } : { hidden: '{!isCTLD}' },
 		allowBlank: false,
 		fieldLabel: gettext('Target portal group')
 	    }
